foursight_core/react/api/react_api.py

```diff
@@ -597,27 +664,255 @@ def reactapi_aws_s3_buckets_keys(self, request: dict, env: str, bucket: str) ->
+            response["portal"] = {}
+            if not portal_url:
+                return None
+            response["portal"]["url"] = get_portal_base_url(portal_url)
+            response["portal"]["health_url"] = response["portal"]["url"] + "/health?format=json"
+            response["portal"]["health_ui_url"] = response["portal"]["url"] + "/health"
+            portal_health_response = requests.get(response["portal"]["health_url"])
+            if portal_health_response.status_code != 200:
+                response["portal"]["error"] = "Cannot fetch Portal health URL."
+                response["portal"]["health_url_status"] = portal_health_response.status_code
+                return None
+            portal_health_json = portal_health_response.json()
+            response["portal"]["versions"] = {
+                "portal": portal_health_json.get("project_version"),
+                "snovault": portal_health_json.get("snovault_version"),
+                "dcicutils": portal_health_json.get("utils_version"),
+                "python": portal_health_json.get("python_version")
+            }
+            portal_uptime = portal_health_json.get("uptime")
+            portal_started = convert_uptime_to_datetime(portal_uptime)
+            response["portal"]["started"] = convert_utc_datetime_to_useastern_datetime_string(portal_started)
+            response["portal"]["identity"] = portal_health_json.get("identity")
+            response["portal"]["elasticsearch"] = portal_health_json.get("elasticsearch")
+            response["portal"]["database"] = portal_health_json.get("database")
+            response["portal"]["health"] = portal_health_json
+            foursight_url = get_foursight_base_url(portal_health_json.get("foursight"))
+            response["portal"]["foursight_url"] = foursight_url
+            return foursight_url
+
+        ignored(request)
+        response = {"accounts_file": self._get_accounts_file(), "accounts_file_from_s3": self._get_accounts_file_from_s3()}
+        accounts = self._get_accounts() if not from_s3 else self._get_accounts_from_s3()
+        if not accounts:
+            return self.create_success_response({"status": "No accounts file support."})
+        account = [account for account in accounts if is_account_name_match(account, name)] if accounts else None
+        if not account or len(account) > 1:
+            response["name"] = name
+            response["error"] = f"Cannot find {' unique' if len(account) > 1 else ''} account."
+            return self.create_success_response(response)
+        else:
+            account = account[0]
+
+        response["name"] = account.get("name")
+        response["stage"] = account.get("stage")
+        response["id"] = account.get("name") + (":" + account.get("stage") if account.get("stage") else "")
+
+        account_foursight_url = account.get("foursight_url")
+        account_portal_url = account.get("portal_url")
+
+        if account_foursight_url:
```

Comment:
Is this stuff just checking on general principles or is there an expected use case where these would be missing?

Reply:
I wanted to be extra with what I get back from the Portal health URL as wasn't sure what I could definitively count on.